core/libcamera_app.cpp

```diff
@@ -419,7 +420,9 @@ void LibcameraApp::ConfigureVideo(unsigned int flags)
 			throw std::runtime_error("Low res image larger than video");
 		configuration_->at(lores_index).pixelFormat = libcamera::formats::YUV420;
 		configuration_->at(lores_index).size = lores_size;
-		configuration_->at(lores_index).bufferCount = configuration_->at(0).bufferCount;
+		if(options_->viewfinder_buffer_count > 0)
+			configuration_->at(lores_index).bufferCount = options_->viewfinder_buffer_count;
+		
```

Comment:
This is for the low res image - different from viewfinder.  I think you need to set the viewfinder buffer count a few lines above - ``configuration_->at(0).bufferCount = ...``.